src/services/layerHelper.js

```diff
@@ -0,0 +1,316 @@
+    capabilitiesURL, layerName) {
+  var parser = new ol.format.WMTSCapabilities();
+  var layer = new ol.layer.Tile();
+  var deferred = this.$q_.defer();
+  this.setHelperID(layer, capabilitiesURL, layerName);
+
+  this.$http_.get(capabilitiesURL).then(function(response) {
+    var result;
+    if (response.data) {
+      result = parser.read(response.data);
+    }
+    if (goog.isDef(result)) {
+      var options = ol.source.WMTS.optionsFromCapabilities(result,
+          {layer: layerName, requestEncoding: 'REST'});
+      layer.setSource(new ol.source.WMTS(options));
+      deferred.resolve(layer);
+    } else {
+      deferred.resolve();
+    }
+  }, function(response) {
+    deferred.resolve();
+  });
+
+  return deferred.promise;
+};
+
+
+/**
+ * Create and return an ol.layer.Group. You can pass a collection of layers to
+ * directly add them in the returned group.
+ * @param {ol.Collection.<ol.layer.Base>=} opt_layers The layer to add to the
+ * returned Group.
+ * @return {ol.layer.Group}
+ * @export
+ */
+ngeo.LayerHelper.prototype.createBasicGroup = function(opt_layers) {
+  var group = new ol.layer.Group();
+  if (goog.isDefAndNotNull(opt_layers)) {
+    group.setLayers(opt_layers);
+  }
+  return group;
+};
+
+
+/**
+ * Get the position of the layer in the map's layers array or -1 if the layer
+ * can't be found.
+ * The layer must be taged by a helperID (return -1 else).
+ * @param {ol.layer.Layer} layer The concerned layer.
+ * @return {number} index or -1 if not found.
+ * @export
+ */
+ngeo.LayerHelper.prototype.getLayerIndex = function(layer) {
+  var layerId = layer.get(this.helperID);
+  if (!goog.isDef(layerId)) {
+    return -1;
+  }
+
+  var i, layers = /** @type {Array.<ol.layer.Layer>} */
+      (this.map_.getLayers().getArray());
+  for (i = 0; i < layers.length; i++) {
+    if (layers[i].get(this.helperID) === layerId) {
+      return i;
+    }
+  }
+  return -1;
+};
+
+
+/**
+ * Retrieve a layer from the given array of layers and based on the given
+ * layers's helperID. Return null if no layer match.
+ * @param {Array.<ol.layer.Layer>} layers An array of layers.
+ * @param {string} layerID an ID like one created by the setHelperID methode
+ *   in this service
+ * @return {ol.layer.Layer?} layer or null;
+ * @export
+ */
+ngeo.LayerHelper.prototype.findLayer = function(layers, layerID) {
+  var i, layer;
+  for (i = 0; i < layers.length; i++) {
+    layer = layers[i];
+    if (layer.get(this.helperID) === layerID) {
+      return layer;
+    }
+  }
+  return null;
+};
+
+
+/**
+ * Add the given layer on the map if it doesn't already exists and if the layer
+ * has a source.
+ * The layer must be taged by a helperID.
+ * @param {ol.layer.Layer} layer The concerned layer.
+ * @return {boolean} true if added, false else.
+ * @export
+ */
+ngeo.LayerHelper.prototype.addLayerToMap = function(layer) {
+  var added = false;
+  if (goog.isDefAndNotNull(layer.getSource())) {
```

Comment:
`if (layer.getSource())` does the same no ?
I think we want to remove `goog` usage if not needed.